Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenViewController.mm

```diff
@@ -340,31 +357,56 @@ - (void)loadView
     ALLOW_DEPRECATED_DECLARATIONS_BEGIN
     [_pipButton setAdjustsImageWhenHighlighted:NO];
     ALLOW_DEPRECATED_DECLARATIONS_END
-    [_pipButton setExtrinsicContentSize:CGSizeMake(60.0, 47.0)];
-    UIImage *startPiPImage = [UIImage imageNamed:@"StartPictureInPictureButton" inBundle:bundle compatibleWithTraitCollection:nil];
-    UIImage *stopPiPImage = [UIImage imageNamed:@"StopPictureInPictureButton" inBundle:bundle compatibleWithTraitCollection:nil];
+    [_pipButton setExtrinsicContentSize:buttonSize];
+    
+    
     [_pipButton setImage:[startPiPImage imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] forState:UIControlStateNormal];
     [_pipButton setImage:[stopPiPImage imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] forState:UIControlStateSelected];
     [_pipButton setTintColor:[UIColor whiteColor]];
     [_pipButton sizeToFit];
     [_pipButton addTarget:self action:@selector(_togglePiPAction:) forControlEvents:UIControlEventTouchUpInside];
 
-    _stackView = adoptNS([[WKFullscreenStackView alloc] init]);
+    if (self._webView._page->preferences().alternateFullScreenControlDesignEnabled()) {
+        
+        UIButtonConfiguration *cancelButtonConfiguration = [UIButtonConfiguration filledButtonConfiguration];
+        // FIXME: this color specification should not be necessary.
+        cancelButtonConfiguration.baseBackgroundColor = [UIColor colorWithWhite:1.0 alpha:0.15];
+        [_cancelButton setConfiguration:cancelButtonConfiguration];
+        
+        UIButtonConfiguration *pipButtonConfiguration = [UIButtonConfiguration filledButtonConfiguration];
+        // FIXME: this color specification should not be necessary.
+        pipButtonConfiguration.baseBackgroundColor = [UIColor colorWithWhite:1.0 alpha:0.15];
+        [_pipButton setConfiguration:pipButtonConfiguration];
+        
+        _stackView = adoptNS([[UIStackView alloc] init]);
+        [_stackView addArrangedSubview:_cancelButton.get()];
+        [_stackView addArrangedSubview:_pipButton.get()];
+        [_stackView setSpacing:24.0];
+    } else {
+        RetainPtr<WKFullscreenStackView> stackView = adoptNS([[WKFullscreenStackView alloc] init]);
+        [stackView addArrangedSubview:_cancelButton.get() applyingMaterialStyle:AVBackgroundViewMaterialStyleSecondary tintEffectStyle:AVBackgroundViewTintEffectStyleSecondary];
+        [stackView addArrangedSubview:_pipButton.get() applyingMaterialStyle:AVBackgroundViewMaterialStylePrimary tintEffectStyle:AVBackgroundViewTintEffectStyleSecondary];
+        _stackView = WTFMove(stackView);
```

Comment:
Nit: above you adopt the view directly into `_stackView`, does this need to be different?

Reply:
It's different so she can use the `WKFullscreenStackView` methods (the one with `applyingMaterialStyle` etc.), since now the member is just the superclass (UIStackView) which doesn't have those.